packages/instantsearch.js/src/lib/utils/__tests__/hydrateSearchClient-test.ts

```diff
@@ -0,0 +1,74 @@
+import { hydrateSearchClient } from '../hydrateSearchClient';
+
+import type { SearchClient, InitialResults } from '../../../types';
+
+describe('hydrateSearchClient', () => {
```

Comment:
let's add another test for all those cases to see that no actual network request happens? could spy on XMLHttpRequest probably?

Reply:
I added a test in `InstantSearchSSRProvider` which is used for both Next.js implems, where I instantiate a search client with a mocked requester and it works well